x-pack/plugins/siem/public/components/page/network/domains_table/first_last_seen/index.test.tsx

```diff
@@ -0,0 +1,211 @@
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 18:35:45.064</span>'
+    );
+  });
+
+  test('Last Seen is empty but not First Seen', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.lastSeen = null;
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={badDateTime} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="first-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+
+    await wait();
+
+    expect(container.innerHTML).toBe(
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 16:09:40.692</span>'
+    );
+  });
+
+  test('First Seen With a bad date time string', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.firstSeen = 'something-invalid';
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={badDateTime} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="first-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+    await wait();
+    expect(container.innerHTML).toBe('something-invalid');
+  });
+
+  test('Last Seen With a bad date time string', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.lastSeen = 'something-invalid';
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={badDateTime} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="last-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+    await wait();
+    expect(container.innerHTML).toBe('something-invalid');
+  });
+
+  test('Show error message', async () => {
+    const myErrorMock = cloneDeep(mockFirstLastSeenDomainQuery);
+    delete myErrorMock[0].result;
+    myErrorMock[0].result = {
+      errors: [{ message: 'Error!' }],
+    };
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={myErrorMock} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="last-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+    await wait();
+    expect(container.innerHTML).toBe('GraphQL error: Error!');
```

Comment:
These tests sometimes fail but more importantly this test is asserting that this below is what will happen:

<img width="1554" alt="Screen Shot 2019-04-11 at 4 55 29 PM" src="https://user-images.githubusercontent.com/1151048/56063834-a2ef1600-5d2d-11e9-9ec2-f0dcc7e68afc.png">

And I think we want an error toaster/pop up and either we try again to get the value (again and again and again) ... or we give up and just display a `--`. But putting the error string in the UI might cause users to think that is a real value or they could do weird things like drag and drop errors onto timeline.

Reply:
Definitely agree, thanks! I refactored this test to return the empty value `--` on error and also added support for Toasts to both the `FirstLastSeenHost` & `FirstLastSeenDomain` hooks. Looks like when you use an explicit `<ApolloConsumer>` for your query, the default configured `errorLink` (which normally handles the write to redux for building toasts) is not used, so you have to specify your own.
<p align="center">
<img src="https://user-images.githubusercontent.com/2946766/56084774-4e16d280-5df5-11e9-81b5-76fdbf351653.png" width="300"></p>

Reply:
This was decision made during the meeting, that we should show the error and not like we did not get any

Reply:
And I do not think that value was draggable anyway